tools/depcheck/pkg/cmd/trace.go

```diff
@@ -0,0 +1,251 @@
+	return nil
+}
+
+func ensureEntrypointPrefix(entrypoints []string, prefix string) []string {
+	for idx, entry := range entrypoints {
+		if strings.HasPrefix(entry, prefix) {
+			continue
+		}
+
+		entrypoints[idx] = path.Join(prefix, entry)
+	}
+
+	return entrypoints
+}
+
+func ensureVendorEntrypoint(entrypoints []string, prefix string) []string {
+	hasVendor := false
+	for _, entry := range entrypoints {
+		if strings.HasSuffix(path.Clean(entry), "vendor") {
+			hasVendor = true
+			break
+		}
+	}
+	if !hasVendor {
+		vendor := ensureEntrypointPrefix([]string{"vendor"}, prefix)
+		vendor = ensureEntrypointEllipsis(vendor)
+		entrypoints = append(entrypoints, vendor[0])
+	}
+
+	return entrypoints
+}
+
+// ensureEntrypointEllipsis receives a list of paths
+// and ensures that each path ends in an ellipsis (...).
+func ensureEntrypointEllipsis(entypoints []string) []string {
+	parsedRoots := []string{}
+	for _, entry := range entypoints {
+		if strings.HasSuffix(entry, "...") {
+			parsedRoots = append(parsedRoots, entry)
+			continue
+		}
+
+		slash := ""
+		if string(entry[len(entry)-1]) != "/" {
+			slash = "/"
+		}
+		entry = strings.Join([]string{entry, slash, "..."}, "")
+		parsedRoots = append(parsedRoots, entry)
+	}
+
+	return parsedRoots
+}
+
+func (o *TraceOptions) Validate() error {
+	if len(o.GraphOptions.Entrypoints) == 0 {
+		return errors.New("at least one root package must be provided")
+	}
+	if len(o.OutputFormat) > 0 && o.OutputFormat != "dot" {
+		return fmt.Errorf("invalid output format provided: %s", o.OutputFormat)
+	}
+
+	return nil
+}
+
+// Run execs `go list` on all package entrypoints specified through --root.
+// Each package's ImportPath and non-transitive (immediate) imports are
+// filtered and aggregated. A package is filtered based on whether its ImportPath
+// matches an accepted pattern defined in the set of validPackages.
+// Each aggregated package becomes a node in a generated dependency graph.
+// An edge is created between a package and each of its Imports.
+func (o *TraceOptions) Run() error {
+	pkgs, err := depgraph.CalculatePackages(
+		ensureEntrypointEllipsis(
+			ensureVendorEntrypoint(o.GraphOptions.Entrypoints, o.GraphOptions.RepoRoot),
+		),
+	)
+	if err != nil {
+		return err
+	}
+
+	g, err := depgraph.BuildGraph(pkgs, o.GraphOptions.Entrypoints, o.GraphOptions.Excludes)
+	if err != nil {
+		return err
+	}
+
+	if len(o.GraphOptions.Filters) > 0 {
+		g, err = depgraph.FilterPackages(g, o.GraphOptions.Filters)
+		if err != nil {
+			return err
+		}
+	}
+
+	// possibleRoots are nodeNames that should be excluded
+	// from pruning or from analysis. We cannot rely on our
+	// given entrypoints as our source of roots because
+	// nodes created for package entrypoints are likely
+	// to have incomplete edge data based on the .go files
+	// immediately present in an entrypoint directory, and
+	// the direct import paths that they define.
```

Comment:
a big comment that isn't very helpful.  Move this to a function, use long line lengths, and provide an example.